zipline/assets/_assets.pyx

```diff
@@ -187,37 +201,58 @@ cdef class Asset:
         """
         return cls(**dict_)
 
-    def _is_alive(self, dt, bool normalized):
+    def _is_alive_for_session(self, session_label):
```

Comment:
Looks like each call site for this function calls `session_for_minute` immediately before.  Should we move that into here?